src/common/intrusive_lru.h

```diff
@@ -170,6 +172,22 @@ class intrusive_lru {
     }
   }
 
+  /*
+   * Clears all elements
+   */
+  void clear() {
+    for (auto i = lru_set.begin(); i != lru_set.end(); i++) {
+      if ((*i).lru) {
+	mark_as_unreferenced(*i, false);
```

Comment:
I think the idea of having a referenced element (with use_count > 0) marked as un-referenced is not intuitive.

Reply:
It's not actually UB, btw.  The entry won't be in the LRU, but it will still be live.  Still, if any of these actually are unreferences after flush(), there's a problem.  Do we know where these references are?

Reply:
mark_as_unreferenced adds *i to the unreferenced_list.  We clear the lru_set but not the unreferenced_list, and intrusive_ptr_release below won't take care of it when the element's use_count eventually goes to 0.  Eventually, both intrusive_ptr_release and evict will try to delete the object leading to UB both in the form of a read-after-delete and a double-delete.